### PR TITLE
Automating Code Formatting in PyGuide.md

### PR DESCRIPTION
**Pull request by AlexanderDokuchaev**
### Changes

Update PyGuide.md
 - added `Automating Code Formatting` paragraph 
 - removed paragraphs about code style that will be auto-formatted by 'black' and 'isort'

 
### Reason for changes

https://github.com/openvinotoolkit/nncf/pull/1692

### Related tickets

CVS-74733

### CODE REVIEW

**Comment by alexsu52** [comment]
run openvino pre-commit tests

File: docs/styleguide/PyGuide.md [hunk 1]
```diff
@@ -264,263 +239,54 @@ class C(ABC):
...
+<a id="4-python-style-rules"></a>
 <a id="python-style-rules"></a>
-## 3 Python Style Rules 
-
-<a id="s3.1-line-length"></a>
-<a id="31-line-length"></a>
-<a id="line-length"></a>
-### 3.1 Line length 
-
-Maximum line length is *120 characters*.
```
Comment (alexsu52) [inline docs/styleguide/PyGuide.md:308]: Maximum line length is the input parameter for black. So I recommend to fix it in the style guide.

File: docs/styleguide/PyGuide.md [hunk 2]
```diff
@@ -264,263 +239,54 @@ class C(ABC):
...
-
-Explicit exceptions to the 120 character limit:
-
--   Long import statements.
```
Comment (alexsu52) [inline docs/styleguide/PyGuide.md:312]: It looks like this list is a list of exceptions when we can disable auto-formatting.

File: docs/styleguide/PyGuide.md [hunk 3]
```diff
@@ -264,263 +239,54 @@ class C(ABC):
...
 Be sure to use the right style for module, function, method docstrings and
 inline comments.
 
-<a id="s3.5.1-modules"></a>
-<a id="351-modules"></a>
+<a id="s4.1.1-modules"></a>
+<a id="411-modules"></a>
 <a id="modules"></a>
-#### 3.5.1 Modules 
+#### 1.1.1 Modules
```
Comment (alexsu52) [inline docs/styleguide/PyGuide.md:289]:
```suggestion
#### 4.1.1 Modules
```

**Review by alexsu52 — CHANGES_REQUESTED** [review]

File: docs/styleguide/PyGuide.md (hunk 1 above)
Comment (AlexanderDokuchaev) [reply docs/styleguide/PyGuide.md:308]: Black parameters contains in `pyproject.toml`, and isort and black use it. 
Also `pre-commit run` will reformate code if run tools with another configuration, in result pre-commit job will fall.

I think will be better if collect only rules that will not be check/change automatically.

File: docs/styleguide/PyGuide.md (hunk 2 above)
Comment (AlexanderDokuchaev) [reply docs/styleguide/PyGuide.md:312]: I assume that isort will only be disabled for cases where autoformatting breaks something like in the case of `konia`.

I will add cases when disabling the formatter is acceptable.

File: docs/styleguide/PyGuide.md [hunk 4]
```diff
@@ -4,159 +4,134 @@
...
+<a id="auto-code-formatting"></a>
+## 2 Automating Code Formatting
+
+To maintain consistency and readability throughout the codebase, we use the [black](https://github.com/psf/black)
+and [isort](https://github.com/PyCQA/isort) tools for formatting. Before committing any changes,
+it's important to run a pre-commit command to ensure that the code is properly formatted.
+You can use the following commands for this:
+
+```bash
+pre-commit run     # Check and format only staged files
```
[the remark this answers is not in the document]
Comment (AlexanderDokuchaev) [reply docs/styleguide/PyGuide.md]: Added `make pre-commit`

File: docs/styleguide/PyGuide.md [hunk 5]
```diff
@@ -4,159 +4,134 @@
...
+pre-commit run -a  # Check and format all files
+```
+
+Also recommend configuring your IDE to run Black and isort tools automatically when saving files.
+
+Automatic code formatting is mandatory for all Python files, but you can disable it for specific cases if required.
```
[the remark this answers is not in the document]
Comment (AlexanderDokuchaev) [reply docs/styleguide/PyGuide.md]: Added

**Comment by ksilligan** [comment]
run openvino pre-commit tests

**Comment by AlexanderDokuchaev** [comment]
run openvino pre-commit tests

**Comment by alexsu52** [comment]
run openvino pre-commit tests

**Pull request merged**